File: SearchAlgorithms/FibonacciSearch/FibonacciSearch.cs
using System;

class FibonacciSearch
{
    static int FibonacciSearchAlgorithm(int[] arr, int target)
    {
        int n = arr.Length;
        int[] fibNumbers = GenerateFibonacciNumbers(n);
        int offset = -1;

        while (fibNumbers[fibNumbers.Length - 1] > 1)
        {
            int i = MinOf(offset + fibNumbers[fibNumbers.Length - 2], n - 1);

            if (arr[i] < target)
            {
                fibNumbers = ResizeArray(fibNumbers, fibNumbers.Length - 2);
                offset = i;
            }
            else if (arr[i] > target)
            {
                fibNumbers = ResizeArray(fibNumbers, fibNumbers.Length - 1);
            }
            else
            {
                return i;
            }
        }

        if (fibNumbers[fibNumbers.Length - 1] == 1 && arr[offset + 1] == target)
        {
            return offset + 1;
        }

        return -1;
    }

    static int[] GenerateFibonacciNumbers(int n)
    {
        var fibNumbers = new int[] { 0, 1 };

        while (fibNumbers[fibNumbers.Length - 1] < n)
        {
            int nextFib = fibNumbers[fibNumbers.Length - 1] + fibNumbers[fibNumbers.Length - 2];
            Array.Resize(ref fibNumbers, fibNumbers.Length + 1);
            fibNumbers[fibNumbers.Length - 1] = nextFib;
        }

        return fibNumbers;
    }

    static int MinOf(int a, int b)
    {
        return a < b ? a : b;
    }

    static int[] ResizeArray(int[] arr, int newSize)
    {
        Array.Resize(ref arr, newSize);
        return arr;
    }

File: SearchAlgorithms/FibonacciSearch/FibonacciSearch.py
def fibonacci_search(arr, target):
    def generate_fibonacci_numbers(n):
        fib_numbers = [0, 1]
        while fib_numbers[-1] < n:
            fib_numbers.append(fib_numbers[-1] + fib_numbers[-2])
        return fib_numbers

    def min_of(a, b):
        return a if a < b else b

    n = len(arr)
    fib_numbers = generate_fibonacci_numbers(n)
    offset = -1

    while fib_numbers[-1] > 1:
        i = min_of(offset + fib_numbers[len(fib_numbers) - 2], n - 1)

        if arr[i] < target:
            fib_numbers = fib_numbers[:len(fib_numbers) - 2]
            offset = i
        elif arr[i] > target:
            fib_numbers = fib_numbers[:len(fib_numbers) - 1]
        else:
            return i

    if fib_numbers[-1] == 1 and arr[offset + 1] == target:
        return offset + 1

    return -1